app/models/client.rb

```diff
@@ -312,18 +355,19 @@ def make_unavailable_in_all_routes
   end
 
   def unavailable(permanent:false, contact_id:nil)
-    active_match = client_opportunity_matches.active.first
     Client.transaction do
-      if active_match.present?
-        opportunity = active_match.opportunity
-        active_match.canceled!
-        MatchEvents::Parked.create!(
-            match_id: active_match.id,
-            contact_id: contact_id
-        )
-        opportunity.update(available_candidate: true)
-        Matching::RunEngineJob.perform_later
+
+      active_matches do |active_match|
+        active_match.canceled!(contact_id)
+        active_match.opportunity.update(available_candidate: true)
       end
+
+      Matching::RunEngineJob.perform_later
+      MatchEvents::Parked.create!(
```

Comment:
This can be written as: `parked_events.create!(contact_id: contact_id)` to take advantage of the relationship.

Reply:
ok